src/core/mac/sub_mac.cpp

```diff
@@ -254,43 +253,28 @@ Error SubMac::Receive(uint8_t aChannel)
 }
 
 #if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
-Error SubMac::CslSample(uint8_t aPanChannel)
+void SubMac::CslSample(void)
 {
     Error error = kErrorNone;
 
-    if (!IsCslChannelSpecified())
-    {
-        mCslChannel = aPanChannel;
-    }
-
 #if OPENTHREAD_CONFIG_MAC_FILTER_ENABLE
     VerifyOrExit(!mRadioFilterEnabled, error = Get<Radio>().Sleep());
 #endif
 
-    switch (mCslState)
-    {
-    case kCslSample:
-        error = Get<Radio>().Receive(mCslChannel);
-        break;
-    case kCslSleep:
+    SetState(kStateCslSample);
+
 #if !OPENTHREAD_CONFIG_MAC_CSL_DEBUG_ENABLE
-        error = Get<Radio>().Sleep(); // Don't actually sleep for debugging
+    IgnoreError(Get<Radio>().Sleep()); // Don't actually sleep for debugging
 #endif
-        break;
-    case kCslIdle:
-        ExitNow(error = kErrorInvalidState);
-    default:
-        OT_ASSERT(false);
-    }
-
-    SetState(kStateCslSample);
 
 exit:
+#if OPENTHREAD_CONFIG_MAC_FILTER_ENABLE
     if (error != kErrorNone)
     {
         LogWarn("CslSample() failed, error: %s", ErrorToString(error));
     }
-    return error;
+#endif
```

Comment:
How about if remove the logging of `error` here?
I don't see any useful value in  loging error under `OPENTHREAD_CONFIG_MAC_FILTER_ENABLE` case.

Reply:
Agree.